chain/consensus/hierarchical/subnet/resolver/resolver.go

```diff
@@ -454,19 +551,69 @@ func (r *Resolver) ResolveCrossMsgs(ctx context.Context, c cid.Cid, from address
 
 }
 
+func (r *Resolver) WaitLockedStateResolved(ctx context.Context, c cid.Cid, from address.SubnetID, actor address.Address) chan error {
+	out := make(chan error)
```

Comment:
We should use a buffered channel. This allows the goroutine to exit if we don't collect the error.
```out := make(chan error, 1)```

Reply:
Good catch, thanks.